xkb/xkb.c

```diff
@@ -1521,8 +1504,7 @@ ProcXkbGetMap(ClientPtr client)
     if (rpcbuf.error)
         return BadAlloc;
```

Comment:
This check could've been done right after the `XkbAssembleMap`, before the swaps. Not a big deal though and not directly related to these change.

Reply:
fixed.